test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureEnvironmentTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.FunctionalTests
+{
+    public class TestingInfrastructureEnvironmentTests
+    { 
+        [Fact]
+        public void TestingInfrastructure_DefaultsEnvironmentToDevelopment()
+        {
+            var factory = new DefaultEnvironmentWebApplicationFactory();
+            var client = factory.CreateClient();
+            Assert.Equal(EnvironmentName.Development, factory.EnvironmentName);
+        }
+
+        [Fact]
+        public void TestingInfrastructure_PreservesEnvironmentSetByVariable()
+        {
+            var factory = new StagingEnvironmentWebApplicationFactory();
+            var client = factory.CreateClient();
+            Assert.Equal(EnvironmentName.Staging, factory.EnvironmentName);
+        }
+    }
+
+    public class DefaultEnvironmentWebApplicationFactory : WebApplicationFactory<BasicWebSite.Startup>
```

Comment:
Make these classes `private`

Reply:
Oh, and move them inside `TestingInfrastructureEnvironmentTests`

Reply:
OK, will do.